### ISSUE TITLE
[4.5.0] Pause at height, redo layers inconsistent

### ISSUE THREAD

**Issue by azhural** [Ultimaker/Cura#7517]
**Application version**
4.5.0

**Platform**
Ubuntu 18.04

**Printer**
Creality Ender-3

**Reproduction steps**
1. Use Pause at height
2. Set to layers, choose any layer
3. Set redo layers to 10, 1 or 0.
4. Slice

**Actual results**
1. Redo layers: 10
At layer 50 it moves to Z10.2, does the pause, then prints layer 40 at 10.2. Then moves to 8.4 to do layer 41.
```
;LAYER:49
;TYPE:WALL-INNER
;MESH:xyzCalibration_cube.stl
G1 F1500 X108.5 Y126.5 E750.57301
G1 X108.5 Y108.5 E751.17169
G1 X114.995 Y108.5 E751.38772
G1 X115.001 Y110.397 E751.45081
G1 X118.273 Y110.397 E751.55964
G1 X118.882 Y109.767 E751.58878
G1 X118.884 Y108.5 E751.63092
G1 X126.5 Y108.5 E751.88423
G1 X126.5 Y114.858 E752.0957
G1 X124.564 Y114.859 E752.16009
G1 X124.563 Y119.326 E752.30867
G1 X126.5 Y119.33 E752.37309
G1 X126.5 Y126.5 E752.61157
G0 F9000 X126.9 Y126.9
G1 F1500 X108.1 Y126.9 E753.23686
G1 X108.1 Y108.1 E753.86215
G1 X115.394 Y108.1 E754.10475
G1 X115.4 Y109.997 E754.16784
G1 X118.103 Y109.997 E754.25774
G1 X118.482 Y109.605 E754.27588
G1 X118.484 Y108.1 E754.32593
G1 X126.9 Y108.1 E754.60585
G1 X126.9 Y115.258 E754.84393
G1 X124.964 Y115.259 E754.90832
G1 X124.963 Y118.927 E755.03032
G1 X126.9 Y118.931 E755.09474
G1 X126.9 Y126.9 E755.35979
G0 F9000 X127.3 Y127.3
;TYPE:WALL-OUTER
G1 F1500 X107.7 Y127.3 E756.01169
G1 X107.7 Y107.7 E756.66359
G1 X115.793 Y107.7 E756.93276
G1 X115.799 Y109.597 E756.99586
G1 X117.933 Y109.597 E757.06684
G1 X118.082 Y109.443 E757.07396
G1 X118.084 Y107.7 E757.13194
G1 X127.3 Y107.7 E757.43846
G1 X127.3 Y115.658 E757.70315
G1 X125.364 Y115.659 E757.76754
G1 X125.363 Y118.528 E757.86296
G1 X127.3 Y118.532 E757.92739
G1 X127.3 Y127.3 E758.21901
G0 F9000 X127.03 Y127.03
G0 X126.419 Y124.904
;TYPE:FILL
G1 F3000 X110.094 Y108.579 E758.98689
G0 F9000 X112.569 Y108.578
G1 F3000 X108.579 Y123.467 E759.49957
G0 F9000 X108.579 Y124.035
G1 F3000 X110.963 Y126.419 E759.61171
G0 F9000 X111.53 Y126.419
G1 F3000 X126.418 Y122.43 E760.12435
G0 F9000 X124.482 Y116.737
G1 F3000 X108.579 Y120.998 E760.67194
G0 F9000 X108.579 Y115.55
G1 F3000 X119.448 Y126.419 E761.18319
G0 F9000 X120.212 Y126.419
G1 F3000 X124.992 Y108.579 E761.79748
G0 F9000 X126.418 Y110.006
G1 F3000 X108.579 Y114.786 E762.41174
G0 F9000 X118.272 Y110.476
G1 F3000 X114 Y126.42 E762.96074
G0 F9000 X124.778 Y114.778
G1 F3000 X118.961 Y108.961 E763.23436
;MESH:NONMESH
G0 F300 X118.961 Y108.961 Z10.2
G0 F9000 X126.5 Y126.5
;TIME_ELAPSED:1017.072565
;TYPE:CUSTOM
;added code by post processing
;script: PauseAtHeight.py
;current layer: 50
M83 ; switch to relative E values for any needed retraction
G1 F3000 E-25
G1 F300 Z11.2 ; move up a millimeter to get out of the way
G1 F9000 X20 Y20
G1 F300 Z15 ; too close to bed--move to at least 15mm
M104 S25 ; standby temperature
M0 ; Do the actual pause
M109 S200 ; resume temperature
G1 F3000 E25
G1 F3000 E-25
G1 F300 Z11.2
G1 F9000 X108.5 Y126.5
G1 F300 Z10.2 ; move back down to resume height
G1 F3000 E25
G1 F3000 ; restore extrusion feedrate
M82 ; switch back to absolute E values
G92 E629.95363
;LAYER:40
;TYPE:WALL-INNER
;MESH:xyzCalibration_cube.stl
G1 F1500 X108.5 Y126.5 E629.95363
G1 X108.5 Y108.5 E630.55231
G1 X113.994 Y108.5 E630.73504
G1 X113.995 Y110.396 E630.7981
G1 X120.02 Y110.396 E630.9985
G1 X120.022 Y108.5 E631.06156
G1 X126.5 Y108.5 E631.27702
G1 X126.5 Y115.281 E631.50255
G1 X124.561 Y115.281 E631.56705
G1 X124.561 Y118.961 E631.68944
G1 X126.5 Y118.962 E631.75393
G1 X126.5 Y126.5 E632.00465
G0 F9000 X126.9 Y126.9
G1 F1500 X108.1 Y126.9 E632.62994
G1 X108.1 Y108.1 E633.25523
G1 X114.394 Y108.1 E633.46457
G1 X114.395 Y109.996 E633.52763
G1 X119.62 Y109.996 E633.70141
G1 X119.622 Y108.1 E633.76448
G1 X126.9 Y108.1 E634.00654
G1 X126.9 Y115.681 E634.25869
G1 X124.961 Y115.681 E634.32318
G1 X124.961 Y118.561 E634.41897
G1 X126.9 Y118.562 E634.48346
G1 X126.9 Y126.9 E634.76078
G0 F9000 X127.3 Y127.3
;TYPE:WALL-OUTER
G1 F1500 X107.7 Y127.3 E635.41268
G1 X107.7 Y107.7 E636.06458
G1 X114.794 Y107.7 E636.30053
G1 X114.795 Y109.596 E636.36359
G1 X116.971 Y109.595 E636.43596
G1 X116.973 Y109.212 E636.4487
G1 X116.971 Y107.7 E636.49899
G1 X117.142 Y107.7 E636.50468
G1 X117.141 Y109.595 E636.56771
G1 X119.22 Y109.596 E636.63685
G1 X119.222 Y107.7 E636.69992
G1 X127.3 Y107.7 E636.96859
G1 X127.3 Y116.081 E637.24734
G1 X125.361 Y116.081 E637.31184
G1 X125.361 Y118.161 E637.38102
G1 X127.3 Y118.162 E637.44551
G1 X127.3 Y127.3 E637.74944
G0 F9000 X127.03 Y127.03
G0 X126.133 Y126.419
;TYPE:FILL
G1 F3000 X108.579 Y108.865 E638.57513
G0 F9000 X108.579 Y113.468
G1 F3000 X119.752 Y110.475 E638.95984
G0 F9000 X119.59 Y110.474
G1 F3000 X115.318 Y126.42 E639.50891
G0 F9000 X117.648 Y126.419
G1 F3000 X108.579 Y117.35 E639.93549
G0 F9000 X108.579 Y119.68
G1 F3000 X124.48 Y115.419 E640.48302
G0 F9000 X124.48 Y115.409
G1 F3000 X121.53 Y126.419 E640.86213
G0 F9000 X126.419 Y121.111
G1 F3000 X108.58 Y125.892 E641.4764
G0 F9000 X108.579 Y125.836
G1 F3000 X109.163 Y126.419 E641.50384
G0 F9000 X109.106 Y126.419
G1 F3000 X113.887 Y108.578 E642.11817
G0 F9000 X114.567 Y109.824
G0 X118.675 Y110.476
G1 F3000 X124.48 Y116.28 E642.3912
G0 F9000 X124.536 Y115.2
G1 F3000 X126.31 Y108.579 E642.61918
G0 F9000 X126.418 Y108.689
G1 F3000 X120.1 Y110.381 E642.83673
G0 F9000 X125.265 Y108.58
G1 F3000 X126.419 Y109.734 E642.89101
G0 F9000 X126.419 Y114.9
G1 F3000 X125.298 Y115.2 E642.9296
;MESH:NONMESH
G0 F300 X125.298 Y115.2 Z8.4
G0 F9000 X125.133 Y115.853
G0 X125.133 Y118.389
G0 X126.5 Y126.5
;TIME_ELAPSED:866.317990
;LAYER:41
;TYPE:WALL-INNER
```

2. Redo layers: 1
Redoes layer 49 on top of layer 49 (Z10, Z10.2), then does layer 50 inside the second layer 49 (Z10.2)
```
G0 F300 X119.085 Y109.286 Z10
G0 F9000 X126.5 Y126.5
;TIME_ELAPSED:1000.433048
;LAYER:49
;TYPE:WALL-INNER
;MESH:xyzCalibration_cube.stl
G1 F1500 X108.5 Y126.5 E750.57301
G1 X108.5 Y108.5 E751.17169
G1 X114.995 Y108.5 E751.38772
G1 X115.001 Y110.397 E751.45081
G1 X118.273 Y110.397 E751.55964
G1 X118.882 Y109.767 E751.58878
G1 X118.884 Y108.5 E751.63092
G1 X126.5 Y108.5 E751.88423
G1 X126.5 Y114.858 E752.0957
G1 X124.564 Y114.859 E752.16009
G1 X124.563 Y119.326 E752.30867
G1 X126.5 Y119.33 E752.37309
G1 X126.5 Y126.5 E752.61157
G0 F9000 X126.9 Y126.9
G1 F1500 X108.1 Y126.9 E753.23686
G1 X108.1 Y108.1 E753.86215
G1 X115.394 Y108.1 E754.10475
G1 X115.4 Y109.997 E754.16784
G1 X118.103 Y109.997 E754.25774
G1 X118.482 Y109.605 E754.27588
G1 X118.484 Y108.1 E754.32593
G1 X126.9 Y108.1 E754.60585
G1 X126.9 Y115.258 E754.84393
G1 X124.964 Y115.259 E754.90832
G1 X124.963 Y118.927 E755.03032
G1 X126.9 Y118.931 E755.09474
G1 X126.9 Y126.9 E755.35979
G0 F9000 X127.3 Y127.3
;TYPE:WALL-OUTER
G1 F1500 X107.7 Y127.3 E756.01169
G1 X107.7 Y107.7 E756.66359
G1 X115.793 Y107.7 E756.93276
G1 X115.799 Y109.597 E756.99586
G1 X117.933 Y109.597 E757.06684
G1 X118.082 Y109.443 E757.07396
G1 X118.084 Y107.7 E757.13194
G1 X127.3 Y107.7 E757.43846
G1 X127.3 Y115.658 E757.70315
G1 X125.364 Y115.659 E757.76754
G1 X125.363 Y118.528 E757.86296
G1 X127.3 Y118.532 E757.92739
G1 X127.3 Y127.3 E758.21901
G0 F9000 X127.03 Y127.03
G0 X126.419 Y124.904
;TYPE:FILL
G1 F3000 X110.094 Y108.579 E758.98689
G0 F9000 X112.569 Y108.578
G1 F3000 X108.579 Y123.467 E759.49957
G0 F9000 X108.579 Y124.035
G1 F3000 X110.963 Y126.419 E759.61171
G0 F9000 X111.53 Y126.419
G1 F3000 X126.418 Y122.43 E760.12435
G0 F9000 X124.482 Y116.737
G1 F3000 X108.579 Y120.998 E760.67194
G0 F9000 X108.579 Y115.55
G1 F3000 X119.448 Y126.419 E761.18319
G0 F9000 X120.212 Y126.419
G1 F3000 X124.992 Y108.579 E761.79748
G0 F9000 X126.418 Y110.006
G1 F3000 X108.579 Y114.786 E762.41174
G0 F9000 X118.272 Y110.476
G1 F3000 X114 Y126.42 E762.96074
G0 F9000 X124.778 Y114.778
G1 F3000 X118.961 Y108.961 E763.23436
;MESH:NONMESH
G0 F300 X118.961 Y108.961 Z10.2
G0 F9000 X126.5 Y126.5
;TIME_ELAPSED:1017.072565
;TYPE:CUSTOM
;added code by post processing
;script: PauseAtHeight.py
;current layer: 50
M83 ; switch to relative E values for any needed retraction
G1 F3000 E-25
G1 F300 Z11.2 ; move up a millimeter to get out of the way
G1 F9000 X20 Y20
G1 F300 Z15 ; too close to bed--move to at least 15mm
M104 S25 ; standby temperature
M0 ; Do the actual pause
M109 S200 ; resume temperature
G1 F3000 E25
G1 F3000 E-25
G1 F300 Z11.2
G1 F9000 X108.5 Y126.5
G1 F300 Z10.2 ; move back down to resume height
G1 F3000 E25
G1 F3000 ; restore extrusion feedrate
M82 ; switch back to absolute E values
G92 E750.57301
;LAYER:49
;TYPE:WALL-INNER
;MESH:xyzCalibration_cube.stl
G1 F1500 X108.5 Y126.5 E750.57301
G1 X108.5 Y108.5 E751.17169
G1 X114.995 Y108.5 E751.38772
G1 X115.001 Y110.397 E751.45081
G1 X118.273 Y110.397 E751.55964
G1 X118.882 Y109.767 E751.58878
G1 X118.884 Y108.5 E751.63092
G1 X126.5 Y108.5 E751.88423
G1 X126.5 Y114.858 E752.0957
G1 X124.564 Y114.859 E752.16009
G1 X124.563 Y119.326 E752.30867
G1 X126.5 Y119.33 E752.37309
G1 X126.5 Y126.5 E752.61157
G0 F9000 X126.9 Y126.9
G1 F1500 X108.1 Y126.9 E753.23686
G1 X108.1 Y108.1 E753.86215
G1 X115.394 Y108.1 E754.10475
G1 X115.4 Y109.997 E754.16784
G1 X118.103 Y109.997 E754.25774
G1 X118.482 Y109.605 E754.27588
G1 X118.484 Y108.1 E754.32593
G1 X126.9 Y108.1 E754.60585
G1 X126.9 Y115.258 E754.84393
G1 X124.964 Y115.259 E754.90832
G1 X124.963 Y118.927 E755.03032
G1 X126.9 Y118.931 E755.09474
G1 X126.9 Y126.9 E755.35979
G0 F9000 X127.3 Y127.3
;TYPE:WALL-OUTER
G1 F1500 X107.7 Y127.3 E756.01169
G1 X107.7 Y107.7 E756.66359
G1 X115.793 Y107.7 E756.93276
G1 X115.799 Y109.597 E756.99586
G1 X117.933 Y109.597 E757.06684
G1 X118.082 Y109.443 E757.07396
G1 X118.084 Y107.7 E757.13194
G1 X127.3 Y107.7 E757.43846
G1 X127.3 Y115.658 E757.70315
G1 X125.364 Y115.659 E757.76754
G1 X125.363 Y118.528 E757.86296
G1 X127.3 Y118.532 E757.92739
G1 X127.3 Y127.3 E758.21901
G0 F9000 X127.03 Y127.03
G0 X126.419 Y124.904
;TYPE:FILL
G1 F3000 X110.094 Y108.579 E758.98689
G0 F9000 X112.569 Y108.578
G1 F3000 X108.579 Y123.467 E759.49957
G0 F9000 X108.579 Y124.035
G1 F3000 X110.963 Y126.419 E759.61171
G0 F9000 X111.53 Y126.419
G1 F3000 X126.418 Y122.43 E760.12435
G0 F9000 X124.482 Y116.737
G1 F3000 X108.579 Y120.998 E760.67194
G0 F9000 X108.579 Y115.55
G1 F3000 X119.448 Y126.419 E761.18319
G0 F9000 X120.212 Y126.419
G1 F3000 X124.992 Y108.579 E761.79748
G0 F9000 X126.418 Y110.006
G1 F3000 X108.579 Y114.786 E762.41174
G0 F9000 X118.272 Y110.476
G1 F3000 X114 Y126.42 E762.96074
G0 F9000 X124.778 Y114.778
G1 F3000 X118.961 Y108.961 E763.23436
;MESH:NONMESH
G0 F300 X118.961 Y108.961 Z10.2
G0 F9000 X126.5 Y126.5
;TIME_ELAPSED:1017.072565
;LAYER:50
;TYPE:WALL-INNER
;MESH:xyzCalibration_cube.stl
G1 F1500 X108.5 Y126.5 E763.83304
G1 X108.5 Y108.5 E764.43172
G1 X115.066 Y108.5 E764.65011
G1 X115.066 Y110.199 E764.70662
G1 X115.066 Y110.398 E764.71323
G1 X115.265 Y110.398 E764.71752
G1 X118.824 Y110.397 E764.83589
G1 X118.846 Y108.5 E764.89899
G1 X126.5 Y108.5 E765.15356
G1 X126.5 Y114.738 E765.36104
G1 X124.565 Y114.74 E765.4254
G1 X124.563 Y119.439 E765.58169
G1 X126.5 Y119.443 E765.64611
G1 X126.5 Y126.5 E765.88083
G0 F9000 X126.9 Y126.9
G1 F1500 X108.1 Y126.9 E766.50612
G1 X108.1 Y108.1 E767.13141
G1 X115.466 Y108.1 E767.3764
G1 X115.466 Y109.799 E767.43291
G1 X115.466 Y109.998 E767.43953
G1 X115.665 Y109.998 E767.44381
G1 X118.429 Y109.997 E767.53574
G1 X118.451 Y108.1 E767.59884
G1 X126.9 Y108.1 E767.87986
G1 X126.9 Y115.138 E768.11394
G1 X124.965 Y115.14 E768.1783
G1 X124.963 Y119.04 E768.30802
G1 X126.9 Y119.044 E768.37244
G1 X126.9 Y126.9 E768.63373
G0 F9000 X127.3 Y127.3
;TYPE:WALL-OUTER
G1 F1500 X107.7 Y127.3 E769.28563
G1 X107.7 Y107.7 E769.93753
G1 X115.866 Y107.7 E770.20913
G1 X115.866 Y109.598 E770.27226
G1 X118.034 Y109.597 E770.34437
G1 X118.056 Y107.7 E770.40747
G1 X127.3 Y107.7 E770.71492
G1 X127.3 Y115.538 E770.97562
G1 X125.365 Y115.54 E771.03997
G1 X125.363 Y118.641 E771.14311
G1 X127.3 Y118.645 E771.20754
G1 X127.3 Y127.3 E771.49541
G0 F9000 X127.03 Y127.03
G0 X126.277 Y126.419
;TYPE:FILL
G1 F3000 X126.419 Y125.89 E771.51362
G0 F9000 X126.419 Y124.705
G1 F3000 X110.294 Y108.58 E772.27209
G0 F9000 X109.109 Y108.579
G1 F3000 X108.579 Y108.721 E772.29034
G0 F9000 X112.423 Y108.579
G1 F3000 X108.579 Y122.923 E772.78426
G0 F9000 X108.579 Y123.836
G1 F3000 X111.162 Y126.419 E772.90576
G0 F9000 X112.074 Y126.419
G1 F3000 X126.418 Y122.576 E773.39967
G0 F9000 X124.483 Y116.883
G1 F3000 X108.58 Y121.144 E773.94726
G0 F9000 X108.579 Y115.351
G1 F3000 X119.648 Y126.419 E774.46789
G0 F9000 X120.066 Y126.42
G1 F3000 X124.846 Y108.579 E775.08221
G0 F9000 X126.419 Y110.152
G1 F3000 X108.579 Y114.932 E775.6965
G0 F9000 X118.126 Y110.475
G1 F3000 X113.854 Y126.419 E776.24551
G0 F9000 X124.858 Y114.659
G1 F3000 X118.921 Y108.721 E776.52479
;MESH:NONMESH
G0 F300 X118.921 Y108.721 Z10.4
G0 F9000 X126.5 Y126.5
;TIME_ELAPSED:1033.987623
;LAYER:51
```

3. Redo layers: 0
Print layer 49 at Z10, pauses, then print layer 50 at Z10.2.
```
G0 F300 X119.085 Y109.286 Z10
G0 F9000 X126.5 Y126.5
;TIME_ELAPSED:1000.433048
;LAYER:49
;TYPE:WALL-INNER
;MESH:xyzCalibration_cube.stl
G1 F1500 X108.5 Y126.5 E750.57301
G1 X108.5 Y108.5 E751.17169
G1 X114.995 Y108.5 E751.38772
G1 X115.001 Y110.397 E751.45081
G1 X118.273 Y110.397 E751.55964
G1 X118.882 Y109.767 E751.58878
G1 X118.884 Y108.5 E751.63092
G1 X126.5 Y108.5 E751.88423
G1 X126.5 Y114.858 E752.0957
G1 X124.564 Y114.859 E752.16009
G1 X124.563 Y119.326 E752.30867
G1 X126.5 Y119.33 E752.37309
G1 X126.5 Y126.5 E752.61157
G0 F9000 X126.9 Y126.9
G1 F1500 X108.1 Y126.9 E753.23686
G1 X108.1 Y108.1 E753.86215
G1 X115.394 Y108.1 E754.10475
G1 X115.4 Y109.997 E754.16784
G1 X118.103 Y109.997 E754.25774
G1 X118.482 Y109.605 E754.27588
G1 X118.484 Y108.1 E754.32593
G1 X126.9 Y108.1 E754.60585
G1 X126.9 Y115.258 E754.84393
G1 X124.964 Y115.259 E754.90832
G1 X124.963 Y118.927 E755.03032
G1 X126.9 Y118.931 E755.09474
G1 X126.9 Y126.9 E755.35979
G0 F9000 X127.3 Y127.3
;TYPE:WALL-OUTER
G1 F1500 X107.7 Y127.3 E756.01169
G1 X107.7 Y107.7 E756.66359
G1 X115.793 Y107.7 E756.93276
G1 X115.799 Y109.597 E756.99586
G1 X117.933 Y109.597 E757.06684
G1 X118.082 Y109.443 E757.07396
G1 X118.084 Y107.7 E757.13194
G1 X127.3 Y107.7 E757.43846
G1 X127.3 Y115.658 E757.70315
G1 X125.364 Y115.659 E757.76754
G1 X125.363 Y118.528 E757.86296
G1 X127.3 Y118.532 E757.92739
G1 X127.3 Y127.3 E758.21901
G0 F9000 X127.03 Y127.03
G0 X126.419 Y124.904
;TYPE:FILL
G1 F3000 X110.094 Y108.579 E758.98689
G0 F9000 X112.569 Y108.578
G1 F3000 X108.579 Y123.467 E759.49957
G0 F9000 X108.579 Y124.035
G1 F3000 X110.963 Y126.419 E759.61171
G0 F9000 X111.53 Y126.419
G1 F3000 X126.418 Y122.43 E760.12435
G0 F9000 X124.482 Y116.737
G1 F3000 X108.579 Y120.998 E760.67194
G0 F9000 X108.579 Y115.55
G1 F3000 X119.448 Y126.419 E761.18319
G0 F9000 X120.212 Y126.419
G1 F3000 X124.992 Y108.579 E761.79748
G0 F9000 X126.418 Y110.006
G1 F3000 X108.579 Y114.786 E762.41174
G0 F9000 X118.272 Y110.476
G1 F3000 X114 Y126.42 E762.96074
G0 F9000 X124.778 Y114.778
G1 F3000 X118.961 Y108.961 E763.23436
;MESH:NONMESH
G0 F300 X118.961 Y108.961 Z10.2
G0 F9000 X126.5 Y126.5
;TIME_ELAPSED:1017.072565
;TYPE:CUSTOM
;added code by post processing
;script: PauseAtHeight.py
;current layer: 50
M83 ; switch to relative E values for any needed retraction
G1 F3000 E-25
G1 F300 Z11.2 ; move up a millimeter to get out of the way
G1 F9000 X20 Y20
G1 F300 Z15 ; too close to bed--move to at least 15mm
M104 S25 ; standby temperature
M0 ; Do the actual pause
M109 S200 ; resume temperature
G1 F3000 E25
G1 F3000 E-25
G1 F300 Z11.2
G1 F9000 X108.5 Y126.5
G1 F300 Z10.2 ; move back down to resume height
G1 F3000 E25
G1 F3000 ; restore extrusion feedrate
M82 ; switch back to absolute E values
G92 E763.23436
;LAYER:50
;TYPE:WALL-INNER
;MESH:xyzCalibration_cube.stl
G1 F1500 X108.5 Y126.5 E763.83304
G1 X108.5 Y108.5 E764.43172
G1 X115.066 Y108.5 E764.65011
G1 X115.066 Y110.199 E764.70662
G1 X115.066 Y110.398 E764.71323
G1 X115.265 Y110.398 E764.71752
G1 X118.824 Y110.397 E764.83589
G1 X118.846 Y108.5 E764.89899
G1 X126.5 Y108.5 E765.15356
G1 X126.5 Y114.738 E765.36104
G1 X124.565 Y114.74 E765.4254
G1 X124.563 Y119.439 E765.58169
G1 X126.5 Y119.443 E765.64611
G1 X126.5 Y126.5 E765.88083
G0 F9000 X126.9 Y126.9
G1 F1500 X108.1 Y126.9 E766.50612
G1 X108.1 Y108.1 E767.13141
G1 X115.466 Y108.1 E767.3764
G1 X115.466 Y109.799 E767.43291
G1 X115.466 Y109.998 E767.43953
G1 X115.665 Y109.998 E767.44381
G1 X118.429 Y109.997 E767.53574
G1 X118.451 Y108.1 E767.59884
G1 X126.9 Y108.1 E767.87986
G1 X126.9 Y115.138 E768.11394
G1 X124.965 Y115.14 E768.1783
G1 X124.963 Y119.04 E768.30802
G1 X126.9 Y119.044 E768.37244
G1 X126.9 Y126.9 E768.63373
G0 F9000 X127.3 Y127.3
;TYPE:WALL-OUTER
G1 F1500 X107.7 Y127.3 E769.28563
G1 X107.7 Y107.7 E769.93753
G1 X115.866 Y107.7 E770.20913
G1 X115.866 Y109.598 E770.27226
G1 X118.034 Y109.597 E770.34437
G1 X118.056 Y107.7 E770.40747
G1 X127.3 Y107.7 E770.71492
G1 X127.3 Y115.538 E770.97562
G1 X125.365 Y115.54 E771.03997
G1 X125.363 Y118.641 E771.14311
G1 X127.3 Y118.645 E771.20754
G1 X127.3 Y127.3 E771.49541
G0 F9000 X127.03 Y127.03
G0 X126.277 Y126.419
;TYPE:FILL
G1 F3000 X126.419 Y125.89 E771.51362
G0 F9000 X126.419 Y124.705
G1 F3000 X110.294 Y108.58 E772.27209
G0 F9000 X109.109 Y108.579
G1 F3000 X108.579 Y108.721 E772.29034
G0 F9000 X112.423 Y108.579
G1 F3000 X108.579 Y122.923 E772.78426
G0 F9000 X108.579 Y123.836
G1 F3000 X111.162 Y126.419 E772.90576
G0 F9000 X112.074 Y126.419
G1 F3000 X126.418 Y122.576 E773.39967
G0 F9000 X124.483 Y116.883
G1 F3000 X108.58 Y121.144 E773.94726
G0 F9000 X108.579 Y115.351
G1 F3000 X119.648 Y126.419 E774.46789
G0 F9000 X120.066 Y126.42
G1 F3000 X124.846 Y108.579 E775.08221
G0 F9000 X126.419 Y110.152
G1 F3000 X108.579 Y114.932 E775.6965
G0 F9000 X118.126 Y110.475
G1 F3000 X113.854 Y126.419 E776.24551
G0 F9000 X124.858 Y114.659
G1 F3000 X118.921 Y108.721 E776.52479
;MESH:NONMESH
G0 F300 X118.921 Y108.721 Z10.4
G0 F9000 X126.5 Y126.5
;TIME_ELAPSED:1033.987623
;LAYER:51
```

**Expected results**
I interpreted "redo" as "print the exact same layer again", which seems to be true from the second redone layer onwards (see redo 10). The first layer redone is at the wrong height imo.

**Comment by Ghostkeeper** [comment]
The intent is that it redoes the last layer before the pause because the act of pausing might bring about some ooze, so you'd get some underextrusion just after restarting the print. You would "prime" by printing one extra layer. This is necessary because other methods of priming are not feasible (priming on the build plate can't be done since the build plate might not be accessible any more, priming on the prime tower might not be done because no prime tower is printed up to the current height, and priming in the infill can't be done because at this stage of processing it's unknown what area is filled with infill).

Why you would ever want to redo more than 1 layer is unknown to me. It seems downright dangerous because the nozzle could clash into the previously printed parts. Should it maybe not make any Z moves at all in the redone part?

**Comment by azhural** [comment]
I was not planning on redoing 10 layers, I was trying to figure out the intended behavior.
The way you described it, a toggle would convey that intention better than a freely selectable number.
Also it's not really an _extra_ layer, as the redone layer and the actual next layer are at the same height.

> Redo a number of previous layers after a pause to increase adhesion.

The tooltip suggests redoing multiple layers might be a usecase (although I can see your point that it probably is not), so I believe my initial point still stands: The actual behavior is all over the place. The first redone layer is added as an extra layer with all the following layers way below inside the already printed part. The part after the redo then starts at the same height as the extra layer.

So if the issue is not "inconsistent behavior" then it should be "UI does not convey intention".

**Comment by Sophist-UK** [comment]
I would propose the following changes:

1. UI should have a checkbox for Redo current layer rather than a numeric box.

2. It seems to me that the redo of the current layer should be more like an iron (using the same settings i.e. slower flow, avoid outer wall. The current layer has already been extruded, so you don't want to try to extrude a full layer of filament, but instead to extrude a much smaller amount that will be pressed firmly into the current layer.

**Comment by Ghostkeeper** [comment]
We've implemented point no. 1 for the next release. With that, the issue here is hidden and no longer in scope.

We did not implement Sophist-UK's second suggestion though. The purpose is not to flatten the surface of the previous layer, but essentially to:

1. Prime the nozzle after being paused and hot and oozing for a while.
2. Ensure sufficient connection after the pause. Pausing often causes a fault line, due to the material cooling down too much and getting time to warp. For some materials it's better then to overextrude a bit to serve as a glue layer.
